calculator.js

```diff
@@ -0,0 +1,119 @@
+(function() {
+
+    var inputField = document.getElementById('screen');
+    var currNumber = 0;
+    var result = 0;
+
+    var buttons = document.querySelector('.allButtons');
+    var numberButtons = buttons.querySelectorAll('.number');
+    /*=======================================================================*/
+
+    function Screen() {
+
+        this.element = inputField;
+    }
+
+    var screen = new Screen();
+
+    Screen.prototype.setNumber = function() {
+
+        this.element = (function() {
+
+            for (var i = 0; i <= numberButtons.length - 1; i++) {
```

Comment:
You missed the point, Screen (and especially) setNumber function should not know anything about number buttons etc. *Screen should server only as an interface for display and nothing else.* Your screen contains too much logic. Here is the skeleton, take it from here:
```
function Screen() {
	this._element = document.getElementById('screen');
}

Screen.prototype.setNumber = function(number) {
	this._element.value = number;
}

Screen.prototype.addDigit = function(digit) {
	var currValue = parseInt(this._element.value);
	currValue = currValue * 10 + digit;
	this.setNumber(currValue);
}
```